src/Toml/PrefixTree.hs

```diff
@@ -95,11 +96,10 @@ type PrefixMap a = HashMap Piece (PrefixTree a)
 
 -- | Data structure to represent table tree for @toml@.
 data PrefixTree a
+      -- | last piece of a key, and value at that point.
     = Leaf !Key !a
-    | Branch { bCommonPref :: !Prefix         -- ^ greatest common prefix
-             , bVal        :: !(Maybe a)      -- ^ value by key = prefix
-             , bPrefixMap  :: !(PrefixMap a)  -- ^ suffixes of prefix
-             }
+      -- | greatest common prefix, optional value at that point and values of suffixes.
+    | Branch !Prefix !(Maybe a) !(PrefixMap a)
```

Comment:
I guess (I'm not sure though) that it's possible to write comments to fields without records, something like:
```haskell
   | Branch 
             !Prefix         -- ^ greatest common prefix
             !(Maybe a)      -- ^ value by key = prefix
             !(PrefixMap a)  -- ^ suffixes of prefix
```

Reply:
Heh, that's sad, I'll check if they have an issue for this feature request.

Reply:
@vrom911 @jiegillet There's a way to document fields of constructor. But you also need to document constructor itself. See this PR:

* https://github.com/haskell/haddock/pull/709

Reply:
It's only available in Haddock 2.21.0, but stack is using 2.20.0. Is that something we can specify in one of the config files?

Reply:
@jiegillet It's okay if this is not supported by `stack`. I'm uploading packages to Hackage using `cabal new-*` commands, so we shouldn't expect any problems with this :+1: